src/dawn/CodeGen/GridTools/GTCodeGen.cpp

```diff
@@ -793,32 +796,47 @@ void GTCodeGen::generateStencilClasses(
         // is resolved
         // https://github.com/MeteoSwiss-APN/dawn/issues/110
         // ===-----------------------------------------------------------------------------------===
-        for(int dim = 0; dim < ext.getSize() - 1; ++dim) {
-          std::string at_call = "template at<" + std::to_string(dim) + ">()";
-          std::string storage = codeGenProperties.getParamType(stencilInstantiation, fieldInfo);
-          // assert for + accesses
-          // ===---------------------------------------------------------------------------------===
-          // PRODUCTIONTODO: [STAGGERING]
-          // we need the staggering offset in K in order to have valid production code
-          // https://github.com/MeteoSwiss-APN/dawn/issues/108
-          // ===---------------------------------------------------------------------------------===
-          std::string staggeringoffset = (dim == 2) ? " - 1" : "";
-          StencilConstructor.addStatement(
-              "static_assert((static_cast<int>(" + storage + "::storage_info_t::halo_t::" +
-              at_call + ") >= " + std::to_string(ext[dim].Plus) + staggeringoffset + ") || " + "(" +
-              storage + "::storage_info_t::layout_t::" + at_call + " == -1)," +
-              "\"Used extents exceed halo limits.\")");
-          // assert for - accesses
-          StencilConstructor.addStatement("static_assert(((-1)*static_cast<int>(" + storage +
-                                          "::storage_info_t::halo_t::" + at_call + ") <= " +
-                                          std::to_string(ext[dim].Minus) + ") || " + "(" + storage +
-                                          "::storage_info_t::layout_t::" + at_call + " == -1)," +
-                                          "\"Used extents exceed halo limits.\")");
+        std::string storage = codeGenProperties.getParamType(stencilInstantiation, fieldInfo);
+        if(fieldsExtsMap.find(storage) == fieldsExtsMap.end()) {
+          fieldsExtsMap.insert(std::make_pair(storage, ext));
```

Comment:
`insert` and `ctor` are replaced in c++11 with `emplace`

Reply:
done